keybase/report.go

```diff
@@ -31,6 +31,22 @@ func (c context) reportError(updateErr updater.Error, options updater.UpdateOpti
 	return c.report(data, uri, timeout)
 }
 
+// ReportAction notifies the API server of a client updater action
+func (c context) ReportAction(action updater.UpdateAction, options updater.UpdateOptions) {
+	if err := c.reportAction(action, options, defaultEndpoints.action, time.Minute); err != nil {
+		c.log.Warningf("Error notifying about an action (%s): %s", action, err)
+	}
+}
+
+func (c context) reportAction(action updater.UpdateAction, options updater.UpdateOptions, uri string, timeout time.Duration) error {
+	data := url.Values{}
+	data.Add("install_id", options.InstallID)
+	data.Add("version", options.Version)
+	data.Add("upd_version", options.UpdaterVersion)
+	data.Add("action", action.String())
+	return c.report(data, uri, timeout)
```

Comment:
there is a minute timeout on all reporting requests.. but I think yeah.. the reporting actions shouldn't slow down an update... I hesitate to introduce async behavior here only because it's harder to reason about it